shared/controller-api/src/main/java/io/pravega/shared/controller/tracing/RPCTracingHelpers.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.shared.controller.tracing;
+
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.ForwardingClientCall;
+import io.grpc.ForwardingClientCallListener;
+import io.grpc.ForwardingServerCall;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import io.pravega.common.LoggerHelpers;
+import io.pravega.common.tracing.RequestTag;
+import io.pravega.common.tracing.RequestTracker;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * This class contains utilities to intercept RPC calls the server and client sides, as well as to track incoming
+ * requests for intra-component tracing.
+ */
+@Slf4j
+public final class RPCTracingHelpers {
```

Comment:
We should have test coverage for this class. Do you have it somewhere and I missed it?

Reply:
Added.